packages/flutter/lib/src/material/carousel_theme.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'carousel.dart';
+import 'material.dart';
+import 'theme.dart';
+
+/// Defines default property values for descendant [CarouselView] widgets.
+///
+/// Descendant widgets obtain the current [CarouselViewThemeData] object using
+/// `CarouselViewTheme.of(context)`. Instances of [CarouselViewThemeData] can be
+/// customized with [CarouselViewThemeData.copyWith].
+///
+/// Typically a [CarouselViewThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.carouselViewTheme].
+///
+/// All [CarouselViewThemeData] properties are `null` by default. When null, the [CarouselView]
+/// will provide its own defaults.
+///
+/// See also:
+///
+///  * [CarouselViewTheme], an [InheritedWidget] that propagates the theme to its descendants.
+///  * [ThemeData], which describes the overall theme information for the application.
+@immutable
+class CarouselViewThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.carouselViewTheme].
+  const CarouselViewThemeData({
+    this.elevation,
+    this.backgroundColor,
+    this.overlayColor,
+    this.shape,
+    this.padding,
+  });
+
+  /// The amount of space to surround each carousel item with.
+  ///
+  /// Overrides the default value for [CarouselView.padding].
+  final EdgeInsets? padding;
+
+  /// The background color for each carousel item.
+  ///
+  /// Overrides the default value for [CarouselView.backgroundColor].
+  final Color? backgroundColor;
+
+  /// The z-coordinate at which to place this carousel relative to its parent.
```

Comment:
I think this is not accurate. The elevation is for each carousel item instead of the whole carousel view.